packages/firebase_vertexai/firebase_vertexai/lib/src/vertex_schema.dart

```diff
@@ -0,0 +1,207 @@
+          description: description,
+          nullable: nullable,
+        );
+
+  /// Construct a schema for an array of values with a specified type.
+  Schema.array({
+    required Schema items,
+    String? description,
+    bool? nullable,
+  }) : this(
+          SchemaType.array,
+          description: description,
+          nullable: nullable,
+          items: items,
+        );
+
+  /// Construct a schema for bool value.
+  Schema.boolean({
+    String? description,
+    bool? nullable,
+  }) : this(
+          SchemaType.boolean,
+          description: description,
+          nullable: nullable,
+        );
+
+  /// Construct a schema for an integer number.
+  ///
+  /// The [format] may be "int32" or "int64".
+  Schema.integer({
+    String? description,
+    bool? nullable,
+    String? format,
+  }) : this(
+          SchemaType.integer,
+          description: description,
+          nullable: nullable,
+          format: format,
+        );
+
+  /// Construct a schema for a non-integer number.
+  ///
+  /// The [format] may be "float" or "double".
+  Schema.number({
+    String? description,
+    bool? nullable,
+    String? format,
+  }) : this(
+          SchemaType.number,
+          description: description,
+          nullable: nullable,
+          format: format,
+        );
+
+  /// Construct a schema for String value with enumerated possible values.
+  Schema.enumString({
+    required List<String> enumValues,
+    String? description,
+    bool? nullable,
+  }) : this(
+          SchemaType.string,
+          enumValues: enumValues,
+          description: description,
+          nullable: nullable,
+          format: 'enum',
+        );
+
+  /// Construct a schema for a String value.
+  Schema.string({
+    String? description,
+    bool? nullable,
+    String? format,
+  }) : this(SchemaType.string,
+            description: description, nullable: nullable, format: format);
+
+  /// The type of this value.
+  SchemaType type;
+
+  /// The format of the data.
+  ///
+  /// This is used only for primitive datatypes.
+  ///
+  /// Supported formats:
+  ///  for [SchemaType.number] type: float, double
+  ///  for [SchemaType.integer] type: int32, int64
+  ///  for [SchemaType.string] type: enum. See [enumValues]
+  String? format;
+
+  /// A brief description of the parameter.
+  ///
+  /// This could contain examples of use.
+  /// Parameter description may be formatted as Markdown.
+  String? description;
+
+  /// Whether the value mey be null.
+  bool? nullable;
+
+  /// Possible values if this is a [SchemaType.string] with an enum format.
+  List<String>? enumValues;
+
+  /// Schema for the elements if this is a [SchemaType.array].
+  Schema? items;
+
+  /// Properties of this type if this is a [SchemaType.object].
+  Map<String, Schema>? properties;
+
+  /// The keys from [properties] for properties that are optional if this is a
+  /// [SchemaType.object]. Any properties that's not listed in optional will be
```

Comment:
[nit] add a blank line following the header sentence.
https://dart.dev/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

Reply:
done